reframe/frontend/dependencies.py

```diff
@@ -90,6 +115,8 @@ def resolve_dep(src, dst):
         for v in adjacent:
             v.in_degree += 1
 
+    _dfs(graph)
```

Comment:
I will remove the DFS traversal from here. If the graph has a cycle, it will loop forever. I will set the level inside the `validate_deps()`.